### ISSUE TITLE
Migrate configure-aws-credentials to use AWS SDK for JavaScript (v3)

### ISSUE THREAD

**Issue by GeorgeDavis-TriumphTech** [aws-actions/configure-aws-credentials#680]
### Describe the feature

```
(node:1629) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
```

### Use Case

There is a warning message when we use `aws-actions/configure-aws-credentials@v1-node16` in GitHub Actions workflow.

### Proposed Solution

Migrate configure-aws-credentials to use AWS SDK for JavaScript (v3)

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by povisenko** [comment]
+1
aws-actions/configure-aws-credentials@v1-node16

**Comment by ngb7h** [comment]
We're also using aws-actions/configure-aws-credentials@v1-node16, same warning.

**Comment by kellertk** [comment]
The JS v3 SDK no longer supports exporting a global credential object, which is used by our action. A major rewrite of the action and our tests is needed to support v3. This is in progress and will coincide with a v3 release of this action :)

**Comment by TeoTN** [comment]
@kellertk Because of this issue I can't deploy my app, is there any workaround?
I tried adding the following env var but it didn't work:
```
      - uses: aws-actions/configure-aws-credentials@v1-node16
        env:
          AWS_SDK_JS_SUPPRESS_MAINTENANCE_MODE_MESSAGE: "1"
```

**Comment by Haarolean** [comment]
@mfn hmm, thanks for double-checking I was convinced that it's related, I'll try to look into other things.

**Comment by AndrewReid-Journey** [comment]
> It seems like they finally dropped the support for this after the deprecation.
> 
> Without any changes, it stopped working just in a matter of a few days, now we get this:
> 
> ```
> Please migrate your code to use AWS SDK for JavaScript (v3).
> For more information, check the migration guide at https://a.co/7PzMCcy
> (Use `node --trace-warnings ...` to show where the warning was created)
> Error: Credentials could not be loaded, please check your action inputs: Could not load credentials from any providers```
> ```

Interestingly, i'm having the exact same issue as you in some of my workflows now. Weirdly I have some workflows which still run OK (albeit showing the warning message about the maintenance mode plans for 2023) but others where this step is failing now. The only differences I can see between these workflows is that the failing ones are declared to run in a container where as the others are not. I wonder if this is the same scenario in your case?

**Comment by Haarolean** [comment]
@AndrewReid-Journey it turns out, on my end, it was related to a workflow being run on a branch from a fork. That's github's security measure which I forgot about. `pull_request` vs `pull_request_target`.

**Comment by LLBlumire** [comment]
I am having the same issue as @AndrewReid-Journey

```
(node:1598) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
Error: Credentials could not be loaded, please check your action inputs: Could not load credentials from any providers
```

Confusingly for me, I have never set up `configure-aws-credentials` before, and so assumed I must be doing something wrong, all online documentation for that error instructs to set `id-token: "write"`, which I had already done.

This whole situation was very confusing, I suppose I picked a bad night to try and set up aws CD!

**Comment by Artur-Oliveira-Jesus** [comment]
there is another way to set up credencials? because i'm having the same problem...

**Comment by aborza-c** [comment]
> The JS v3 SDK no longer supports exporting a global credential object, which is used by our action. A major rewrite of the action and our tests is needed to support v3. This is in progress and will coincide with a v3 release of this action :)

Is this only an issue with setting up credentials directly? What about using OIDC, is that also affected?

**Comment by kkurczewski** [comment]
@mchittineni, this issue is only about removing unaesthetic warning, error you got is not related/caused by this warning. Your error states:
> Error: Credentials could not be loaded, **please check your action inputs**: Could not load credentials from any providers

So most likely you are passing wrong or missing role/region/other param.

**Comment by andreyFernandoSoares** [comment]
require('aws-sdk/lib/maintenance_mode_message').suppress = true;

**Comment by akshit-speed** [comment]
Is there any update for same ? Due to this we are starting to seen some of the workflows are failing randomly while configuring aws credentials step.

**Comment by jensenbox** [comment]
Happened 2 times in the last 7 days for us. Is there an update on when this will be resolved?

**Comment by omegion** [comment]
Same here, it's happening quite often today.

**Comment by gscho** [comment]
When I follow the instructions in the blog post things work. But I don't understand where they are getting the `1c58a3a8518e8759bf075b76b750d4f2df264fcd` thumbprint from. I get these two from `token.actions.githubusercontent.com` :

* 6938fd4d98bab03faadb97b34396831e3780aea1
* f879abce0008e4eb126e0097e46620f5aaae26ad

Where does this thumbprint in the blog post come from?

**Comment by danopia** [comment]
> Where does this thumbprint in the blog post come from?

For some context, here's the certificate chain that I see for GHA in Google Chrome:

![image](https://github.com/aws-actions/configure-aws-credentials/assets/40628/659879ae-3cf7-4aed-a6ca-9cd22e923bcf)

I believe that you are looking at the last certificate (Github's cert), but for AWS OIDC you generally want the first intermediate, which is the second certificate in the list. This is because the first certificate can't be used and the last certificate is renewed at least annually:

| Subject                          | SHA-1       | Lifespan   | Usage with AWS OIDC                           |
|----------------------------------|-------------|------------|------------------------------------|
| DigiCert Global Root CA          | a8985d3a... | 25 years   | ❌ Thumbprint won't be recognized   |
| DigiCert TLS RSA SHA256 2020 CA1 | 1c58a3a8... | 10 years   | ✅ Recognized & infrequent rotation |
| *.actions.githubusercontent.com  | f879abce... | ~1.3 years | ❌ Needs regular rotation          |

Note that this is all a relic of AWS OIDC being paranoid enough to not use the trusted CA list maintained by OSs and browsers. I'd love to see a "Internet root trust store" option from AWS so we can stop worrying about rotating fingerprints 🤷🏻

**Comment by kellertk** [comment]
Hello everyone, the error that you're seeing related to thumbprints and certificates is not related to JavaScript v3. This is a tracking issue for the v3 migration, which we already have in progress. We work extremely closely with the JavaScript SDK team and migrating this action to the new version will happen before the SDK is fully deprecated. For any discussion related to certificate thumbprints, please see #357.

**Comment by kellertk** [comment]
Just to confirm, the warning you're seeing as a result of the v2 SDK being used is merely cosmetic at this time.

**Comment by matthieuh** [comment]
It seems not cosmetic anymore it makes fail my builds 🙁

**Comment by Blunderchips** [comment]
Our pipeline started failing this morning with something related to this. Not getting a warning but a failed job. Is there a work around for this issue?

![image](https://github.com/aws-actions/configure-aws-credentials/assets/7081037/601c8915-a1a8-4901-bcd5-d2c5f709f12a)

**Comment by Blunderchips** [comment]
> The error is not due to the SDK version, but something else. The message you see there is just a warning unrelated to the error.
> 
> Please open up another issue with the details of your workflow @Blunderchips

Ok, will do it when I get a chance. Stopped working over the weekend with no changes our side.

**Comment by Blunderchips** [comment]
> i get exact same issue as Blunderchips when trying to setup this github action for the first time, with a very basic workflow

Yes, same error. It worked on the Friday then not on the Monday. Of course, our keys are defined and are still valid.

**Comment by c-ameron** [comment]
> The JS v3 SDK no longer supports exporting a global credential object, which is used by our action. A major rewrite of the action and our tests is needed to support v3. This is in progress and will coincide with a v3 release of this action :)

Thanks for the info @kellertk . Do you have an ETA of when this will be released? And will this include the other aws actions (ie [amazon-ecs-deploy-task-definition](https://github.com/aws-actions/amazon-ecs-deploy-task-definition))?

**Comment by DanielMcAssey** [comment]
Same here, worked on Friday, stopped working on Monday, nothing changed on our pipeline or AWS

**Comment by kellertk** [comment]
Closed by #791
